Realm/Realm/Helpers/SerializationHelper.cs

```diff
@@ -36,6 +36,7 @@ static SerializationHelper()
         {
             var decimalSerializer = new DecimalSerializer(BsonType.Decimal128, new RepresentationConverter(allowOverflow: false, allowTruncation: false));
             BsonSerializer.RegisterSerializer(decimalSerializer);
+            BsonSerializer.RegisterSerializer(new GuidSerializer(GuidRepresentation.Standard));
```

Comment:
This serializes uuids as type 4 rather than type 0. Cloud expects 4 and the MongoDB.Bson package defaults to 0 for legacy reasons.